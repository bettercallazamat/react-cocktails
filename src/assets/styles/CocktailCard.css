.CocktailCard {
  width: 240px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  transition: 0.3s;
  padding: 10px;
  margin: 10px;
  text-decoration: none;
}

.CocktailCard:hover {
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2), 0 12px 40px 0 rgba(0, 0, 0, 0.19);
}

.CocktailCard h3 {
  font-size: 18px;
  margin: 8px 0 10px 0;
  color: #000;
}

.CocktailCard img {
  width: 100%;
  object-fit: cover;
}

.CocktailCard span {
  display: inline-block;
  padding: 4px;
  margin: 0 4px 4px 0;
  border-radius: 2px;
  font-size: 12px;
  color: #f2f2f2;
}

/* FOR CONDITIONAL TAGS */

/* ALCOHOLIC SECTION */
.alcoholic {
  background-color: #8a34d9;
}

.nonAlcoholic {
  background-color: #65b891;
}

.optionalAlcohol {
  background-color: #fa8334;
}

/* CATEGORY SECTION */
.ordinaryDrink {
  background-color: #429ea6;
}

.cocktail {
  background-color: #e3655b;
}

.milk {
  background-color: #d0c88e;
}

.other {
  background-color: #80a4ed;
}

.cocoa {
  background-color: #5f464b;
}

.shot {
  background-color: #053225;
}

.coffeeTea {
  background-color: #8e4a49;
}

.homemadeLiqueur {
  background-color: #593959;
}

.punch {
  background-color: #311847;
}

.beer {
  background-color: #e09f7d;
}

.softDrink {
  background-color: #ec4067;
}
