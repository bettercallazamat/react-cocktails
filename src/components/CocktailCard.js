import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import '../assets/styles/CocktailCard.css';

const CocktailCard = ({ cocktail }) => (
  <div className="CocktailCard">
    <img src={cocktail.strDrinkThumb} alt="drink" />
    <h3 className="coctail-name">{cocktail.strDrink}</h3>
    <span className={classNames({
      alcoholic: cocktail.strAlcoholic === 'Alcoholic',
      nonAlcoholic: cocktail.strAlcoholic === 'Non alcoholic',
      optionalAlcohol: cocktail.strAlcoholic === 'Optional alcohol',
    })}
    >
      {cocktail.strAlcoholic}
    </span>
    <span className={classNames({
      ordinaryDrink: cocktail.strCategory === 'Ordinary Drink',
      cocktail: cocktail.strCategory === 'Cocktail',
      milk: cocktail.strCategory === 'Milk / Float / Shake',
      other: cocktail.strCategory === 'Other/Unknown',
      cocoa: cocktail.strCategory === 'Cocoa',
      shot: cocktail.strCategory === 'Shot',
      coffeeTea: cocktail.strCategory === 'Coffee / Tea',
      homemadeLiqueur: cocktail.strCategory === 'Homemade Liqueur',
      punch: cocktail.strCategory === 'Punch / Party Drink',
      beer: cocktail.strCategory === 'Beer',
      softDrink: cocktail.strCategory === 'Soft Drink / Soda',
    })}
    >
      {cocktail.strCategory}
    </span>
  </div>
);

CocktailCard.propTypes = {
  cocktail: PropTypes.objectOf(PropTypes.string),
};

CocktailCard.defaultProps = {
  cocktail: {},
};

export default CocktailCard;
