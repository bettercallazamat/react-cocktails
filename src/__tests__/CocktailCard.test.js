import React from 'react';
// import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import CocktailCard from '../components/CocktailCard';

describe('CocktailCard', () => {
  const cocktail = {
    idDrink: '11007',
    strDrink: 'Margarita',
    strCategory: 'Ordinary Drink',
    strAlcoholic: 'Alcoholic',
  };

  it('should match the snapshot', () => {
    const { asFragment } = render(<CocktailCard cocktail={cocktail} />);
    expect(asFragment(<CocktailCard cocktail={cocktail} />)).toMatchSnapshot();
  });
});
