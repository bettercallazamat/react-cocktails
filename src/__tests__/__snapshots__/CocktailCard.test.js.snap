// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CocktailCard should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="CocktailCard"
  >
    <img
      alt="drink"
    />
    <h3
      class="coctail-name"
    >
      Margarita
    </h3>
    <span
      class="alcoholic"
    >
      Alcoholic
    </span>
    <span
      class="ordinaryDrink"
    >
      Ordinary Drink
    </span>
  </div>
</DocumentFragment>
`;
